In [1]:
import selenium
from selenium import webdriver
from selenium.webdriver.common.by import By

In [2]:
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC

In [3]:
website = 'https://www.espncricinfo.com/records/tournament/indian-premier-league-2023-15129'
path = r'C:\Users\abhis\Downloads\ChromeDriver'

In [4]:
driver = webdriver.Chrome(path)

C:\Users\abhis\AppData\Local\Temp\ipykernel_16596\2107036205.py:1: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(path)


In [5]:
driver.get(website)
try:
    WebDriverWait(driver, 20).until(EC.element_to_be_clickable((By.ID, 'wzrk-cancel'))).click()
except:
    print("could not click")
    pass

In [6]:
most_runs_page = WebDriverWait(driver, 1).until(EC.element_to_be_clickable((By.XPATH, '//a[@href="/records/tournament/batting-most-runs-career/indian-premier-league-2023-15129"]'))).click()
most_runs_page

In [7]:
most_runs = driver.find_elements(By.TAG_NAME, 'tr')

for i in most_runs:
    print(i.text)

Player
Mat
Inns
NO
Runs
HS
Ave
BF
SR
100
50
0
4s
6s
F du Plessis (RCB)
14 14 1 730 84 56.15 475 153.68 - 8 - 60 36
Shubman Gill (GT)
15 15 2 722 104* 55.53 484 149.17 2 4 1 71 23
V Kohli (RCB)
14 14 2 639 101* 53.25 457 139.82 2 6 1 65 16
DP Conway (CSK)
15 14 2 625 92* 52.08 456 137.06 - 6 1 73 16
YBK Jaiswal (RR)
14 14 1 625 124 48.07 382 163.61 1 5 1 82 26
RD Gaikwad (CSK)
15 14 1 564 92 43.38 384 146.87 - 4 - 43 29
DA Warner (DC)
14 14 0 516 86 36.85 392 131.63 - 6 2 69 10
SA Yadav (MI)
15* 14 2 511 103* 42.58 276 185.14 1 4 1 56 24
RK Singh (KKR)
14 14 6 474 67* 59.25 317 149.52 - 4 - 31 29
H Klaasen (SRH)
12 11 2 448 104 49.77 253 177.07 1 2 - 32 25
Ishan Kishan (MI)
15* 14 0 439 75 31.35 306 143.46 - 3 - 51 18
N Rana (KKR)
14 14 1 413 75 31.76 293 140.95 - 3 - 39 20
VR Iyer (KKR)
14 14 0 404 104 28.85 277 145.84 1 2 1 32 21
GJ Maxwell (RCB)
14 14 2 400 77 33.33 218 183.48 - 5 2 29 31
JC Buttler (RR)
14 14 0 392 95 28.00 282 139.00 - 4 5 42 14
S Dube (CSK)
15 13 2 386 52 35.09 24

In [8]:
Player = []
Mat = []
Inns = []
NO = []
Runs = []
HS = []
Ave =[]
BF = []
SR = []
Hundred = []
Fifty = []
Duck = []
Four = []
Six =[]

In [9]:
for i in most_runs:
    name = i.find_element(By.XPATH, './td[1]').text
    Player.append(name)
    print(name)
    Mat.append(i.find_element(By.XPATH, './td[2]').text)
    Inns.append(i.find_element(By.XPATH, './td[3]').text)
    NO.append(i.find_element(By.XPATH, './td[4]').text)
    Runs.append(i.find_element(By.XPATH, './td[5]').text)
    HS.append(i.find_element(By.XPATH, './td[6]').text)
    Ave.append(i.find_element(By.XPATH, './td[7]').text)
    BF.append(i.find_element(By.XPATH, './td[8]').text)
    SR.append(i.find_element(By.XPATH, './td[9]').text)
    Hundred.append(i.find_element(By.XPATH, './td[10]').text)
    Fifty.append(i.find_element(By.XPATH, './td[11]').text)
    Duck.append(i.find_element(By.XPATH, './td[12]').text)
    Four.append(i.find_element(By.XPATH, './td[13]').text)
    Six.append(i.find_element(By.XPATH, './td[14]').text)

Player
F du Plessis (RCB)
Shubman Gill (GT)
V Kohli (RCB)
DP Conway (CSK)
YBK Jaiswal (RR)
RD Gaikwad (CSK)
DA Warner (DC)
SA Yadav (MI)
RK Singh (KKR)
H Klaasen (SRH)
Ishan Kishan (MI)
N Rana (KKR)
VR Iyer (KKR)
GJ Maxwell (RCB)
JC Buttler (RR)
S Dube (CSK)
C Green (MI)
S Dhawan (PBKS)
MP Stoinis (LSG)
SV Samson (RR)
KR Mayers (LSG)
N Pooran (LSG)
Prabhsimran Singh (PBKS)
RG Sharma (MI)
JM Sharma (PBKS)
V Shankar (GT)
AM Rahane (CSK)
SO Hetmyer (RR)
WP Saha (GT)
HH Pandya (GT)
JJ Roy (KKR)
AR Patel (DC)
LS Livingstone (PBKS)
SM Curran (PBKS)
KL Rahul (LSG)
NT Tilak Varma (MI)
RA Tripathi (SRH)
MA Agarwal (SRH)
D Padikkal (RR)
DA Miller (GT)
AK Markram (SRH)
A Badoni (LSG)
Rahmanullah Gurbaz (KKR)
AD Russell (KKR)
Abhishek Sharma (SRH)
B Sai Sudharsan (GT)
PD Salt (DC)
TH David (MI)
N Wadhera (MI)
RR Rossouw (DC)
HC Brook (SRH)
A Taide (PBKS)
KH Pandya (LSG)
RA Jadeja (CSK)
Abdul Samad (SRH)
MK Pandey (DC)
M Shahrukh Khan (PBKS)
DC Jurel (RR)
Q de Kock (LSG)
KD Karthik (RCB)
Sikandar R

In [10]:
import pandas as pd
import numpy as np

In [11]:
batsman = pd.DataFrame({'Player': Player[1:], 'Mat': Mat[1:], 'Inns' : Inns[1:], 'NO':NO[1:], 'Runs' : Runs[1:], 'HS':HS[1:],
'Ave': Ave[1:], 'BF' : BF[1:], 'SR':SR[1:], 'Hundred':Hundred[1:], 'Fifty': Fifty[1:], 'Duck':Duck[1:], 'Four':Four[1:], 'Six':Six[1:]})

In [12]:
batsman.head()

Player Mat Inns NO Runs    HS    Ave   BF      SR Hundred  \
0  F du Plessis (RCB)  14   14  1  730    84  56.15  475  153.68       -   
1   Shubman Gill (GT)  15   15  2  722  104*  55.53  484  149.17       2   
2       V Kohli (RCB)  14   14  2  639  101*  53.25  457  139.82       2   
3     DP Conway (CSK)  15   14  2  625   92*  52.08  456  137.06       -   
4    YBK Jaiswal (RR)  14   14  1  625   124  48.07  382  163.61       1   

  Fifty Duck Four Six  
0     8    -   60  36  
1     4    1   71  23  
2     6    1   65  16  
3     6    1   73  16  
4     5    1   82  26

In [13]:
'F du Plessis (RCB)'.split('(')

['F du Plessis ', 'RCB)']

In [14]:
'F du Plessis (RCB)'.split('(')[0]

'F du Plessis '

In [15]:
'F du Plessis (RCB)'.split('(')[1][0:-1]

'RCB'

In [16]:
def get_player(col):
    return col.split('(')[0]

def get_team(col):
    return col.split('(')[1][0:-1]

In [17]:
batsman_data = batsman.copy()

In [18]:
batsman_data['Team'] = batsman_data['Player'].apply(get_team)

In [19]:
batsman_data['Player'] = batsman_data['Player'].apply(get_player)

In [20]:
batsman_data.head()

Player Mat Inns NO Runs    HS    Ave   BF      SR Hundred Fifty  \
0  F du Plessis   14   14  1  730    84  56.15  475  153.68       -     8   
1  Shubman Gill   15   15  2  722  104*  55.53  484  149.17       2     4   
2       V Kohli   14   14  2  639  101*  53.25  457  139.82       2     6   
3     DP Conway   15   14  2  625   92*  52.08  456  137.06       -     6   
4   YBK Jaiswal   14   14  1  625   124  48.07  382  163.61       1     5   

  Duck Four Six Team  
0    -   60  36  RCB  
1    1   71  23   GT  
2    1   65  16  RCB  
3    1   73  16  CSK  
4    1   82  26   RR

In [21]:
batting_order = pd.read_csv(r'D:\Untitled Folder\IPL/batting_order.csv')

In [22]:
batting_order.head()

Player  BO
0  F du Plessis    1
1   YBK Jaiswal    1
2     DP Conway    1
3       V Kohli    1
4    RD Gaikwad    1

In [23]:
batsman_data = pd.merge(left=batsman_data,right=batting_order,on='Player',how='left')

In [24]:
batsman_data.head()

Player Mat Inns NO Runs    HS    Ave   BF      SR Hundred Fifty  \
0  F du Plessis   14   14  1  730    84  56.15  475  153.68       -     8   
1  Shubman Gill   15   15  2  722  104*  55.53  484  149.17       2     4   
2       V Kohli   14   14  2  639  101*  53.25  457  139.82       2     6   
3     DP Conway   15   14  2  625   92*  52.08  456  137.06       -     6   
4   YBK Jaiswal   14   14  1  625   124  48.07  382  163.61       1     5   

  Duck Four Six Team   BO  
0    -   60  36  RCB  1.0  
1    1   71  23   GT  1.0  
2    1   65  16  RCB  1.0  
3    1   73  16  CSK  1.0  
4    1   82  26   RR  1.0

In [25]:
#Replacing - with 0 in data

for i in batsman_data.columns:
    batsman_data[i] = batsman_data[i].replace(['-'],0)

In [26]:
batsman_data.head()

Player Mat Inns NO Runs    HS    Ave   BF      SR Hundred Fifty  \
0  F du Plessis   14   14  1  730    84  56.15  475  153.68       0     8   
1  Shubman Gill   15   15  2  722  104*  55.53  484  149.17       2     4   
2       V Kohli   14   14  2  639  101*  53.25  457  139.82       2     6   
3     DP Conway   15   14  2  625   92*  52.08  456  137.06       0     6   
4   YBK Jaiswal   14   14  1  625   124  48.07  382  163.61       1     5   

  Duck Four Six Team   BO  
0    0   60  36  RCB  1.0  
1    1   71  23   GT  1.0  
2    1   65  16  RCB  1.0  
3    1   73  16  CSK  1.0  
4    1   82  26   RR  1.0

In [79]:
def r_asterisk(col):
    return col.split('*')[0]

In [80]:
batsman_data.columns

Index(['Player', 'Mat', 'Inns', 'NO', 'Runs', 'HS', 'Ave', 'BF', 'SR',
       'Hundred', 'Fifty', 'Duck', 'Four', 'Six', 'Team', 'BO',
       'Boundary_percent', 'Avg_bf'],
      dtype='object')

In [ ]:
for i in batsman_data.columns[1:8]:
    batsman_data[i] = batsman_data[i].apply(r_asterisk)

In [30]:
batsman_data.head(10)

Player Mat Inns NO Runs   HS    Ave   BF      SR Hundred Fifty Duck  \
0  F du Plessis   14   14  1  730   84  56.15  475  153.68       0     8    0   
1  Shubman Gill   15   15  2  722  104  55.53  484  149.17       2     4    1   
2       V Kohli   14   14  2  639  101  53.25  457  139.82       2     6    1   
3     DP Conway   15   14  2  625   92  52.08  456  137.06       0     6    1   
4   YBK Jaiswal   14   14  1  625  124  48.07  382  163.61       1     5    1   
5    RD Gaikwad   15   14  1  564   92  43.38  384  146.87       0     4    0   
6     DA Warner   14   14  0  516   86  36.85  392  131.63       0     6    2   
7      SA Yadav   15   14  2  511  103  42.58  276  185.14       1     4    1   
8      RK Singh   14   14  6  474   67  59.25  317  149.52       0     4    0   
9     H Klaasen   12   11  2  448  104  49.77  253  177.07       1     2    0   

  Four Six Team   BO  
0   60  36  RCB  1.0  
1   71  23   GT  1.0  
2   65  16  RCB  1.0  
3   73  16  CSK  1.0  
4   82  26   RR  1.0  
5   43  29  CSK  1.0  
6   69  10   DC  1.0  
7   56  24   MI  4.0  
8   31  29  KKR  5.0  
9   32  25  SRH  6.0

In [31]:
batsman_data.dtypes

Player      object
Mat         object
Inns        object
NO          object
Runs        object
HS          object
Ave         object
BF          object
SR          object
Hundred     object
Fifty       object
Duck        object
Four        object
Six         object
Team        object
BO         float64
dtype: object

In [32]:
convert_dict = {
    
    'Player': object,
    'Mat' : int,
    'Inns' : int,
    'NO' : int,
    'Runs' : int,
    'HS' : int,
    'Ave' : float,
    'BF' : int,
    'SR' : float,
    'Hundred' : int,
    'Fifty' : int,
    'Duck' : int,
    'Four' : int,
    'Six' : int,
    'Team' : object
}

In [33]:
batsman_data = batsman_data.astype(convert_dict)

In [34]:
batsman_data.dtypes

Player      object
Mat          int32
Inns         int32
NO           int32
Runs         int32
HS           int32
Ave        float64
BF           int32
SR         float64
Hundred      int32
Fifty        int32
Duck         int32
Four         int32
Six          int32
Team        object
BO         float64
dtype: object

In [35]:
batsman_data['Boundary_percent'] = ((batsman_data['Four']*4+batsman_data['Six']*6)/batsman_data['Runs'])*100

In [36]:
batsman_data.head()

Player  Mat  Inns  NO  Runs   HS    Ave   BF      SR  Hundred  \
0  F du Plessis    14    14   1   730   84  56.15  475  153.68        0   
1  Shubman Gill    15    15   2   722  104  55.53  484  149.17        2   
2       V Kohli    14    14   2   639  101  53.25  457  139.82        2   
3     DP Conway    15    14   2   625   92  52.08  456  137.06        0   
4   YBK Jaiswal    14    14   1   625  124  48.07  382  163.61        1   

   Fifty  Duck  Four  Six Team   BO  Boundary_percent  
0      8     0    60   36  RCB  1.0         62.465753  
1      4     1    71   23   GT  1.0         58.448753  
2      6     1    65   16  RCB  1.0         55.712050  
3      6     1    73   16  CSK  1.0         62.080000  
4      5     1    82   26   RR  1.0         77.440000

In [37]:
batsman_data['Avg_bf'] = batsman_data['BF']/batsman_data['Inns']

In [38]:
#TO left align table

from IPython.core.display import HTML
table_css = 'table {align:left;display:block} '
HTML('<style>{}</style>'.format(table_css))

# Strategy For Openers
	
|Parameters | Description | Criteria |
|:----------|:-----------:|:-------- |
|Batting Average|Avg runs scored in an innings|	>40|
|Strike Rate	|No of runs scored per 100 balls	|>140|
|Innings|	Total innings batted	|>5|
|Boundary %|	% of runs scored in boundaries	|>50|
|batting_order| |<=2|


In [39]:
filter1 = batsman_data['Ave']>40
filter2 = batsman_data['SR']>135
filter3 = batsman_data['Inns'] > 7
filter4 = batsman_data['Boundary_percent'] > 50
filter5 = batsman_data['BO']<=2

In [40]:
openers = batsman_data.where(filter1 & filter2 & filter3 & filter4 & filter5)

In [41]:
openers = openers.dropna()

In [42]:
openers.head(3)

Player   Mat  Inns   NO   Runs     HS    Ave     BF      SR  \
0  F du Plessis   14.0  14.0  1.0  730.0   84.0  56.15  475.0  153.68   
1  Shubman Gill   15.0  15.0  2.0  722.0  104.0  55.53  484.0  149.17   
2       V Kohli   14.0  14.0  2.0  639.0  101.0  53.25  457.0  139.82   

   Hundred  Fifty  Duck  Four   Six Team   BO  Boundary_percent     Avg_bf  
0      0.0    8.0   0.0  60.0  36.0  RCB  1.0         62.465753  33.928571  
1      2.0    4.0   1.0  71.0  23.0   GT  1.0         58.448753  32.266667  
2      2.0    6.0   1.0  65.0  16.0  RCB  1.0         55.712050  32.642857

# Strategy For Middle Order
	
|Parameters | Description | Criteria |
|:----------|:-----------:|:-------- |
|Batting Average|Avg runs scored in an innings|	>40|
|Strike Rate	|No of runs scored per 100 balls	|>120|
|Innings|	Total innings batted	|>5|
|Avg_bf|	Avg balls faced in an innings	|>20|
|batting_order| |>2|


In [52]:
filter1 = batsman_data['Ave']>35
filter2 = batsman_data['SR']>120
filter3 = batsman_data['Inns'] > 5
filter4 = batsman_data['Avg_bf']>19
filter5 = batsman_data['BO'] > 2
filter6 = batsman_data['BO'] <=5

In [53]:
middle_order = batsman_data.where(filter1 & filter2 & filter3 & filter4 & filter5 & filter6)

In [54]:
middle_order = middle_order.dropna()
middle_order.sort_values(by = 'BO')

Player   Mat  Inns   NO   Runs     HS    Ave     BF      SR  \
7          SA Yadav   15.0  14.0  2.0  511.0  103.0  42.58  276.0  185.14   
45  B Sai Sudharsan    6.0   6.0  1.0  223.0   62.0  44.60  178.0  125.28   
8          RK Singh   14.0  14.0  6.0  474.0   67.0  59.25  317.0  149.52   
35   NT Tilak Varma   10.0   9.0  3.0  274.0   84.0  45.66  173.0  158.38   

    Hundred  Fifty  Duck  Four   Six Team   BO  Boundary_percent     Avg_bf  
7       1.0    4.0   1.0  56.0  24.0   MI  4.0         72.015656  19.714286  
45      0.0    2.0   0.0  20.0   5.0   GT  4.0         49.327354  29.666667  
8       0.0    4.0   0.0  31.0  29.0  KKR  5.0         62.869198  22.642857  
35      0.0    1.0   0.0  21.0  18.0   MI  5.0         70.072993  19.222222

# Strategy For Finisher/Lower Order
	
|Parameters | Description | Criteria |
|:----------|:-----------:|:-------- |
|Batting Average|Avg runs scored in an innings|	>25|
|Strike Rate	|No of runs scored per 100 balls	|>150|
|Innings|	Total innings batted	|>3|
|Avg_bf|	Avg balls faced in an innings	|>8|
|batting_order| |>5|


In [55]:
filter1 = batsman_data['Ave']>25
filter2 = batsman_data['SR']>150
filter3 = batsman_data['Inns'] > 3
filter4 = batsman_data['Avg_bf']>=8
filter5 = batsman_data['BO']>5

In [56]:
finisher = batsman_data.where(filter1 & filter2 & filter3 & filter4 & filter5)

In [57]:
finisher.dropna(inplace=True)

In [58]:
finisher.head()

Player   Mat  Inns   NO   Runs     HS    Ave     BF      SR  Hundred  \
9    H Klaasen   12.0  11.0  2.0  448.0  104.0  49.77  253.0  177.07      1.0   
21    N Pooran   15.0  14.0  3.0  358.0   62.0  32.54  206.0  173.78      0.0   
27  SO Hetmyer   14.0  13.0  5.0  299.0   56.0  37.37  197.0  151.77      0.0   
47    TH David   15.0  13.0  6.0  216.0   45.0  30.85  130.0  166.15      0.0   

    Fifty  Duck  Four   Six Team   BO  Boundary_percent     Avg_bf  
9     2.0   0.0  32.0  25.0  SRH  6.0         62.053571  23.000000  
21    2.0   1.0  26.0  26.0  LSG  6.0         72.625698  14.714286  
27    1.0   0.0  12.0  23.0   RR  6.0         62.207358  15.153846  
47    0.0   2.0  11.0  15.0   MI  6.0         62.037037  10.000000

In [59]:
openers.to_csv('openers.csv',index=False)
middle_order.to_csv('middle_order.csv',index=False)
finisher.to_csv('finisher.csv',index=False)

# Bowlers

In [60]:
driver = webdriver.Chrome(path)

C:\Users\abhis\AppData\Local\Temp\ipykernel_16596\2107036205.py:1: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(path)


In [61]:
driver.get(website)
try:
    WebDriverWait(driver, 20).until(EC.element_to_be_clickable((By.ID, 'wzrk-cancel'))).click()
except:
    print("could not click")
    pass

In [ ]:
most_wickets_page = WebDriverWait(driver, 10).until(EC.element_to_be_clickable((By.XPATH, '//a[@href="/records/tournament/bowling-most-wickets-career/indian-premier-league-2023-15129"]'))).click()
most_wickets_page

In [64]:
most_wickets = driver.find_elements(By.TAG_NAME, 'tr')

for i in most_wickets:
    print(i.text)

Player
Mat
Inns
Balls
Overs
Mdns
Runs
Wkts
BBI
BBM
Ave
Econ
SR
4
5
Mohammed Shami (GT)
15 15 354 59.0 2 452 26 4/11 4/11 17.38 7.66 13.61 2 -
Rashid Khan (GT)
15 15 360 60.0 - 475 25 4/30 4/30 19.00 7.91 14.40 1 -
YS Chahal (RR)
14 14 317 52.5 - 432 21 4/17 4/17 20.57 8.17 15.09 3 -
TU Deshpande (CSK)
15 15 317 52.5 - 508 21 3/45 3/45 24.19 9.61 15.09 - -
PP Chawla (MI)
15* 14 324 54.0 - 422 20 3/22 3/22 21.10 7.81 16.20 - -
CV Varun (KKR)
14 14 316 52.4 1 429 20 4/15 4/15 21.45 8.14 15.80 1 -
MM Sharma (GT)
12 12 233 38.5 - 315 19 4/28 4/28 16.57 8.11 12.26 2 -
Mohammed Siraj (RCB)
14 14 300 50.0 - 375 19 4/21 4/21 19.73 7.50 15.78 1 -
RA Jadeja (CSK)
15 15 318 53.0 - 393 19 3/20 3/20 20.68 7.41 16.73 - -
M Pathirana (CSK)
11 11 254 42.2 - 327 17 3/15 3/15 19.23 7.72 14.94 - -
Arshdeep Singh (PBKS)
14 14 305 50.5 - 493 17 4/29 4/29 29.00 9.69 17.94 1 -
Ravi Bishnoi (LSG)
15* 13 279 46.3 - 361 16 3/28 3/28 22.56 7.76 17.43 - -
B Kumar (SRH)
14 14 306 51.0 1 425 16 5/30 5/30 26.56 8.33 

In [65]:
Player = []
Mat = []
Inns = []
Balls = []
Overs = []
Mdns = []
Runs = []
Wkts = []
BBI = []
BBM = []
Ave = []
Econ = []
SR = []
Four = []
Fifer = []

In [66]:
for i in most_wickets:
    name = i.find_element(By.XPATH, './td[1]').text
    Player.append(name)
    print(name)
    Mat.append(i.find_element(By.XPATH, './td[2]').text)
    Inns.append(i.find_element(By.XPATH, './td[3]').text)
    Balls.append(i.find_element(By.XPATH, './td[4]').text)
    Overs.append(i.find_element(By.XPATH, './td[5]').text)
    Mdns.append(i.find_element(By.XPATH, './td[6]').text)
    Runs.append(i.find_element(By.XPATH, './td[7]').text)
    Wkts.append(i.find_element(By.XPATH, './td[8]').text)
    BBI.append(i.find_element(By.XPATH, './td[9]').text)
    BBM.append(i.find_element(By.XPATH, './td[10]').text)
    Ave.append(i.find_element(By.XPATH, './td[11]').text)
    Econ.append(i.find_element(By.XPATH, './td[12]').text)
    SR.append(i.find_element(By.XPATH, './td[13]').text)
    Four.append(i.find_element(By.XPATH, './td[14]').text)
    Fifer.append(i.find_element(By.XPATH, './td[15]').text)

Player
Mohammed Shami (GT)
Rashid Khan (GT)
YS Chahal (RR)
TU Deshpande (CSK)
PP Chawla (MI)
CV Varun (KKR)
MM Sharma (GT)
Mohammed Siraj (RCB)
RA Jadeja (CSK)
M Pathirana (CSK)
Arshdeep Singh (PBKS)
Ravi Bishnoi (LSG)
B Kumar (SRH)
Noor Ahmad (GT)
JP Behrendorff (MI)
R Ashwin (RR)
HV Patel (RCB)
TA Boult (RR)
NT Ellis (PBKS)
MR Marsh (DC)
DL Chahar (CSK)
M Markande (SRH)
MA Wood (LSG)
AR Patel (DC)
M Theekshana (CSK)
SP Narine (KKR)
I Sharma (DC)
KV Sharma (RCB)
Yash Thakur (LSG)
M Jansen (SRH)
Suyash Sharma (KKR)
Kuldeep Yadav (DC)
A Nortje (DC)
Sandeep Sharma (RR)
T Natarajan (SRH)
SM Curran (PBKS)
MM Ali (CSK)
Harpreet Brar (PBKS)
WD Parnell (RCB)
V Vyshak (RCB)
PWH de Silva (RCB)
KH Pandya (LSG)
KK Ahmed (DC)
A Madhwal (MI)
A Zampa (RR)
Avesh Khan (LSG)
RD Chahar (PBKS)
A Mishra (LSG)
AD Russell (KKR)
Naveen-ul-Haq (LSG)
RP Meredith (MI)
SN Thakur (KKR)
AS Joseph (GT)
K Rabada (PBKS)
Mukesh Kumar (DC)
MG Bracewell (RCB)
JB Little (GT)
C Green (MI)
MP Stoinis (LSG)
VG Arora (KKR)
H

In [67]:
wickets = pd.DataFrame({'Player': Player[1:], 'Mat': Mat[1:], 'Inns' : Inns[1:], 'Balls':Balls[1:], 'Overs' : Overs[1:], 'Mdns':Mdns[1:],
'Runs': Runs[1:], 'Wkts' : Wkts[1:], 'BBI':BBI[1:], 'BBM':BBM[1:], 'Ave': Ave[1:], 'Econ':Econ[1:], 'SR':SR[1:], 'Four':Four[1:], 'Fifer':Fifer[1:]})

In [68]:
wickets.head()

Player  Mat Inns Balls Overs Mdns Runs Wkts   BBI   BBM  \
0  Mohammed Shami (GT)   15   15   354  59.0    2  452   26  4/11  4/11   
1     Rashid Khan (GT)   15   15   360  60.0    -  475   25  4/30  4/30   
2       YS Chahal (RR)   14   14   317  52.5    -  432   21  4/17  4/17   
3   TU Deshpande (CSK)   15   15   317  52.5    -  508   21  3/45  3/45   
4       PP Chawla (MI)  15*   14   324  54.0    -  422   20  3/22  3/22   

     Ave  Econ     SR Four Fifer  
0  17.38  7.66  13.61    2     -  
1  19.00  7.91  14.40    1     -  
2  20.57  8.17  15.09    3     -  
3  24.19  9.61  15.09    -     -  
4  21.10  7.81  16.20    -     -

In [69]:
wickets.drop(['BBI','BBM'], inplace=True,axis=1)

In [70]:
wickets.head()

Player  Mat Inns Balls Overs Mdns Runs Wkts    Ave  Econ  \
0  Mohammed Shami (GT)   15   15   354  59.0    2  452   26  17.38  7.66   
1     Rashid Khan (GT)   15   15   360  60.0    -  475   25  19.00  7.91   
2       YS Chahal (RR)   14   14   317  52.5    -  432   21  20.57  8.17   
3   TU Deshpande (CSK)   15   15   317  52.5    -  508   21  24.19  9.61   
4       PP Chawla (MI)  15*   14   324  54.0    -  422   20  21.10  7.81   

      SR Four Fifer  
0  13.61    2     -  
1  14.40    1     -  
2  15.09    3     -  
3  15.09    -     -  
4  16.20    -     -

In [71]:
for i in wickets.columns:
    wickets[i] = wickets[i].replace(['-'],0)

In [72]:
wickets.dtypes

Player    object
Mat       object
Inns      object
Balls     object
Overs     object
Mdns      object
Runs      object
Wkts      object
Ave       object
Econ      object
SR        object
Four      object
Fifer     object
dtype: object

In [77]:
convert_dict = {
    
    'Player': object,
    'Mat' : int,
    'Inns' : int,
    'Balls' : int,
    'Overs' : float,
    'Mdns' : int,
    'Runs' : int,
    'Wkts' : int,
    'Ave' : float,
    'Econ' : float,
    'SR' : float,
    'Four' : int,
    'Fifer' : int
}

In [ ]:
for i in wickets.columns[1:8]:
    wickets[i] = wickets[i].apply(r_asterisk)

In [83]:
wickets = wickets.astype(convert_dict)

In [84]:
wickets.dtypes

Player     object
Mat         int32
Inns        int32
Balls       int32
Overs     float64
Mdns        int32
Runs        int32
Wkts        int32
Ave       float64
Econ      float64
SR        float64
Four        int32
Fifer       int32
dtype: object

In [86]:
wickets.head(10)

Player  Mat  Inns  Balls  Overs  Mdns  Runs  Wkts    Ave  \
0   Mohammed Shami (GT)   15    15    354   59.0     2   452    26  17.38   
1      Rashid Khan (GT)   15    15    360   60.0     0   475    25  19.00   
2        YS Chahal (RR)   14    14    317   52.5     0   432    21  20.57   
3    TU Deshpande (CSK)   15    15    317   52.5     0   508    21  24.19   
4        PP Chawla (MI)   15    14    324   54.0     0   422    20  21.10   
5        CV Varun (KKR)   14    14    316   52.4     1   429    20  21.45   
6        MM Sharma (GT)   12    12    233   38.5     0   315    19  16.57   
7  Mohammed Siraj (RCB)   14    14    300   50.0     0   375    19  19.73   
8       RA Jadeja (CSK)   15    15    318   53.0     0   393    19  20.68   
9     M Pathirana (CSK)   11    11    254   42.2     0   327    17  19.23   

   Econ     SR  Four  Fifer  
0  7.66  13.61     2      0  
1  7.91  14.40     1      0  
2  8.17  15.09     3      0  
3  9.61  15.09     0      0  
4  7.81  16.20     0      0  
5  8.14  15.80     1      0  
6  8.11  12.26     2      0  
7  7.50  15.78     1      0  
8  7.41  16.73     0      0  
9  7.72  14.94     0      0

# Strategy For Bowlers
	
|Parameters | Description | Criteria |
|:----------|:-----------:|:-------- |
|Econ|Average runs given per over|	<8|
|Strike Rate	|No. of balls required to take a wicket	|<16|
|Innings|	Total innings batted	|>8|
|Bowling Average|	runs allowed per wicket	|<20|
|Overs| |>20|


In [87]:
filter1 = wickets['Econ']<8
filter2 = wickets['SR']<16
filter3 = wickets['Inns']>5
filter4 = wickets['Ave']<30
filter5 = wickets['Overs']>20

In [88]:
bowlers = wickets.where(filter1 & filter2 & filter3 & filter4 & filter5)

In [89]:
bowlers.dropna(inplace=True)

In [90]:
bowlers.head()

Player   Mat  Inns  Balls  Overs  Mdns   Runs  Wkts    Ave  \
0   Mohammed Shami (GT)  15.0  15.0  354.0   59.0   2.0  452.0  26.0  17.38   
1      Rashid Khan (GT)  15.0  15.0  360.0   60.0   0.0  475.0  25.0  19.00   
7  Mohammed Siraj (RCB)  14.0  14.0  300.0   50.0   0.0  375.0  19.0  19.73   
9     M Pathirana (CSK)  11.0  11.0  254.0   42.2   0.0  327.0  17.0  19.23   

   Econ     SR  Four  Fifer  
0  7.66  13.61   2.0    0.0  
1  7.91  14.40   1.0    0.0  
7  7.50  15.78   1.0    0.0  
9  7.72  14.94   0.0    0.0

In [91]:
bowlers.to_csv('bowlers.csv',index=False)